### PR TITLE
TASK: New SVG icons for the Media Browser

### PR DESCRIPTION
**Pull request by dfeyer**
This change replace the old PNG based icons with more modern SVG icons.
The new icon set support the old design (classic) and a new one (vivid).

By default the icon set is set to the `vivid` version. But you can
change it by editing your settings:

```yaml
Neos:
  Media:
    iconSet:
      path: 'resource://Neos.Media/Public/IconSets/vivid'
      extension: 'svg'
```

This change also improve the performance by checking if the icon file
exists only once per file extensions.

The only requirement for a custom icon set is to have the `blank.svg`
icon to be used when no specific icon has been found.

You need to clear your thumbail when you switch to a new icon set:

```
./flow media:clearthumbnails
```

### CODE REVIEW

**Comment by dfeyer** [comment]
See https://neos-project.slack.com/archives/C050RKUV7/p1533823819000255 for a screenshot .... github does not allow me to upload an image here...

File: Neos.Media/Classes/Domain/Model/FileTypeIcon.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        $icon = $this->getIconPath($this->extension);
+
+        if (!is_file($icon)) {
+            $icon = $this->getIconPath('blank');
+        }
+
+        return $icon;
+    }
+
+    protected function getIconPath(string $name)
```
Comment (daniellienert) [inline Neos.Media/Classes/Domain/Model/FileTypeIcon.php]: A return type would be nice

File: Neos.Media/Classes/Domain/Model/FileTypeIcon.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public function __construct(string $extension)
+    {
+        $this->extension = $extension;
+    }
+
+    public function alt(): string {
+        return $this->extension;
+    }
+
+    public function path(): string
```
Comment (daniellienert) [inline Neos.Media/Classes/Domain/Model/FileTypeIcon.php]: Have you intentionally left the method comments out?

File: Neos.Media/Classes/Domain/Model/FileTypeIcon.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     * @var string
+     */
+    protected $extension;
+
+    public function __construct(string $extension)
+    {
+        $this->extension = $extension;
+    }
+
+    public function alt(): string {
```
Comment (daniellienert) [inline Neos.Media/Classes/Domain/Model/FileTypeIcon.php]: Style CI complains ...

File: Neos.Media/Classes/Domain/Model/FileTypeIcon.php (hunk 2 above)
Comment (dfeyer) [reply Neos.Media/Classes/Domain/Model/FileTypeIcon.php]: I start to hate method comments ;)

**Comment by dfeyer** [comment]
@daniellienert Nitpick fixed, and a small bug fixed too, ready for an other review

**Comment by dfeyer** [comment]
/me editing files on github is risky ;)

**Comment by dfeyer** [comment]
I think this one can be stashed before merging ;)

**Pull request merged**